src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4712,22 +4713,30 @@ int wm_vuldet_set_agents_info(agent_software **agents_software, update_node **up
             continue;
         }
 
+        json_field = cJSON_GetObjectItem(json_agt_info->child, "node_name");
+        if(cJSON_IsString(json_field) && json_field->valuestring != NULL) {
+            if (0 != strcmp(json_field->valuestring, node_name)) {
+                cJSON_Delete(json_agt_info);
+                continue;
+            }
+        }
+
         json_field = cJSON_GetObjectItem(json_agt_info->child, "os_name");
-        if(cJSON_IsString(json_field) && json_field->valuestring != NULL){
+        if(cJSON_IsString(json_field) && json_field->valuestring != NULL) {
             os_name = json_field->valuestring;
         }
 
         json_field = cJSON_GetObjectItem(json_agt_info->child, "os_major");
-        if(cJSON_IsString(json_field) && json_field->valuestring != NULL){
+        if(cJSON_IsString(json_field) && json_field->valuestring != NULL) {
             os_major = json_field->valuestring;
         }
 
         json_field = cJSON_GetObjectItem(json_agt_info->child, "name");
-        if(cJSON_IsString(json_field) && json_field->valuestring != NULL){
+        if(cJSON_IsString(json_field) && json_field->valuestring != NULL) {
             name = json_field->valuestring;
         }
 
-        json_field = cJSON_GetObjectItem(json_agt_info->child, "os_build");
+        json_field = cJSON_GetObjectItem(json_agt_info->child, "os_build") ;
```

Comment:
Code style: **;** should be aside next to the method.

Reply:
Fixed.